### DISCUSSION TITLE
nrf24 wireless audio stream support

### DISCUSSION THREAD

**Discussion by mkoenings** [pschatzmann/arduino-audio-tools#1162] [Ideas]
I was struggling with the ESPNOW protocol, i could never make it work well enough ( low latency and and "clean" audio transmission).
I then tried the good old nrf24 modules (SPI) , they work really well, at least for my purposes ( mono, latency 6.5msec(!), 32k sample rate, 16bit) I guess the ability to set the frequency to the highest values in the 2.4ghz band is part of the trick. ESPNOW is bound to the WIFI channels, which are very crowded in many areas. 
It would be great to have the boilerplate code (setup, loops for read/write readiness)  'wrapped up' in a clean nrf24Stream class. I could transmit audio successfully between teensy 4.1, esp32 and even arduino nano ( the ones with a nrf24 integrated). the 32 bytes payload are not an issue, the thing is so fast that it keeps up with the audio stream. ( 2 Mbyte transmission rate, writefast() with no ack, no CRC)
I am happy to provide example code, I am basically reading from a buffer into the writefast() method (TX), and reading  a buffer from the radio.read()  straight into the I2s_write(), or a buffer in between.
Thanks for the amazing library!

**Comment by pschatzmann** [comment]
Feel free to submit a Pull Request with your Nrf24Stream class and the related examples

**Comment by pschatzmann** [comment]
I added some [DRAFT implementation with the corresonging examples](https://github.com/pschatzmann/arduino-audio-tools/commit/3bd1a1e233c6fa9bd80fe2516332cb83dea3f714).
